flask_rebar/utils/request_utils.py

```diff
@@ -157,15 +165,15 @@ def raise_400_for_marshmallow_errors(
     :raises: errors.BadRequest
     """
     if not errs:
-        return errors.BadRequest(msg=msg)
+        raise errors.BadRequest(msg=msg)
```

Comment:
If there's no errors this function should just return. [That is what it was previously doing before this bug was introduced](https://github.com/plangrid/flask-rebar/blob/6e1d8d9d81a5d314ec162fdda67ee7a9f3accc6c/flask_rebar/utils/request_utils.py#L115-L133).
```suggestion
        return
```

Reply:
I think keeping this change actually makes things better.

This function is designed around `marshmallow.ValidationError`. I think that if we handle a `ValidationError` and this function is called with the props of the validation error, it should raise a `BadRequest`. Intuitively, by naming, and practically, thinking about a web request, this makes sense.

Instead, we have a case where if `errs` is empty/falsy, we don't do anything. The only caller in flask-rebar is `_get_data_or_400` that passes `errs=e.messages_dict` where e is a `ValidationError`. The ([marshmallow src](https://github.com/marshmallow-code/marshmallow/blob/ac13e0d2353fb34431b14d6a541862a6b89f65c9/src/marshmallow/exceptions.py#L51C9-L51C22)) will raise a `TypeError` if the underlying `messages` prop isn't a dict, which tells me that it is, practically speaking, nearly or always set as a dict when raised as the result of `compat.load`, because we don't see this happening in production.

I think this is a case where introducing a bit of breakage in service of correctness is OK.

Reply:
I'm sold, as long as we document in the release notes that this is a bugfix where previously errors were not raised, it's a small enough change I'm on board with making it as a correction.